packages/linter/src/utils/convert-tslint-to-eslint/project-converter.ts

```diff
@@ -150,8 +150,23 @@ export class ProjectConverter {
    * focus on the project-level config conversion.
    */
   async convertRootTSLintConfig(
-    applyPackageSpecificModifications: (json: Linter.Config) => Linter.Config
+    applyPackageSpecificModifications: (json: Linter.Config) => Linter.Config,
+    rootEslintConfigExists?: boolean
   ): Promise<Exclude<GeneratorCallback, void>> {
+    /**
+     * If root eslint already exists, we will not override it with converted tslint
+     * as this might break existing configuration in place. This is the common scenario
+     * when large projects are migrating one project at a time and apply custom
+     * changes to root config in the meantime.
+     *
+     * We warn user of this action in case .eslintrc.json was created accidentally
+     */
+    if (rootEslintConfigExists) {
```

Comment:
Can the `host.exists('.eslintrc.json')` be inlined in here instead of passing it in?

And I think this should move under the next early bail which checks the flag to ignore the tslint config?

Reply:
the `host.exists('.eslintrc.json')` cannot be inlined unfortunately since the previous step `initESLint` generates the file if it doesn't exist, so we wouldn't know if this is a fresh generic one or existing.